src/Faithlife.Diff/DiffUtility.cs

```diff
@@ -14,57 +14,57 @@ public static class DiffUtility
 		/// Finds the differences between the two lists.
 		/// </summary>
 		/// <typeparam name="T">The type of item in the lists.</typeparam>
-		/// <param name="listFirst">The first list.</param>
-		/// <param name="listSecond">The second list.</param>
+		/// <param name="first">The first list.</param>
+		/// <param name="second">The second list.</param>
 		/// <returns>A sequence of pairs that indicate the differences between the lists.</returns>
 		/// <remarks><para>Each pair identifies a range of items in each list. The items before and after each range
 		/// are the same in both lists; the items in each range are the difference. One of the two ranges can be
 		/// empty, but the index of the range still indicates where the missing items are.</para>
 		/// <para>EqualityComparer&lt;T>.Default is used to compare items.</para></remarks>
-		public static IEnumerable<ValueTuple<IndexRange, IndexRange>> FindDifferences<T>(IList<T> listFirst, IList<T> listSecond)
+		public static IReadOnlyList<(IndexRange FirstRange, IndexRange SecondRange)> FindDifferences<T>(IList<T> first, IList<T> second)
 		{
-			return FindDifferences(listFirst, listSecond, null);
+			return FindDifferences(first, second, null);
 		}
 
 		/// <summary>
 		/// Finds the differences between the two lists.
 		/// </summary>
 		/// <typeparam name="T">The type of item in the lists.</typeparam>
-		/// <param name="listFirst">The first list.</param>
-		/// <param name="listSecond">The second list.</param>
+		/// <param name="first">The first list.</param>
+		/// <param name="second">The second list.</param>
 		/// <param name="comparer">The comparer.</param>
 		/// <returns>A sequence of pairs that indicate the differences between the lists.</returns>
 		/// <remarks><para>Each pair identifies a range of items in each list. The items before and after each range
 		/// are the same in both lists; the items in each range are the difference. One of the two ranges can be
 		/// empty, but the index of the range still indicates where the missing items are.</para>
 		/// <para>If comparer is null, EqualityComparer&lt;T>.Default is used.</para></remarks>
-		public static IEnumerable<ValueTuple<IndexRange, IndexRange>> FindDifferences<T>(IList<T> listFirst, IList<T> listSecond,
+		public static IReadOnlyList<(IndexRange FirstRange, IndexRange SecondRange)> FindDifferences<T>(IList<T> first, IList<T> second,
```

Comment:
If `IList<T>` implemented `IReadOnlyList<T>`, I'd recommend changing it. [Alas...](https://stackoverflow.com/a/35940240/23818)

But maybe we should change it anyway, and introduce an easy way to cast/convert an `IList<T>` to an `IReadOnlyList<T>` in `Faithlife.Utility`. `AsReadOnlyList()` extension method?

Reply:
Arguably, `IReadOnlyList<T>` is the better argument type, as the implied contract is that the method won't mutate its arguments. I think it's worth changing for that alone.

The only problem with that is a caller who has an `IList<T>` and can't easily call the method? A new `.AsReadOnlyList` extension method sounds good. (Try to cast to `IReadOnlyList<T>`; failing that, execute `.ToList()`?)

Reply:
Wouldn't a shim be cheaper than `.ToList()`? I guess it's hard to say for sure. An indirection vs. copying the items.

Reply:
It would be uncommon, so maybe the shim isn't worth the effort.

Reply:
Turns out I need a shim for JSON patching in `Faithlife.Json`: `JToken` is `IList<T>` but not `IReadOnlyList<T>`. Adding this in Faithlife/FaithlifeUtility#5.